test/system/255-auto-update.bats

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# Tests for automatically update images for containerized services
+#
+
+load helpers
+
+UNIT_DIR="/usr/lib/systemd/system"
+SNAME_FILE=$BATS_TMPDIR/services
+
+function setup() {
+    skip_if_remote "systemd tests are meaningless over remote"
+    skip_if_rootless
+
+    basic_setup
+}
+
+function teardown() {
+    while read line; do
```

Comment:
This is unnecessary and misleading: `teardown` runs after every `@test`, not once per `.bats` file. From what I see below, `$SNAME_FILE` only gets written to once per test, therefore it can only contain at most one line. A better way to implement this would be:
```bash
setup() {
   ...
   TEST_SERVICE_NAME=...
...
teardown() {
   if [[ -n "$TEST_SERVICE_NAME" ]]; then
      systemctl disable --now $TEST_SERVICE_NAME
      rm -f $UNIT_DIR/$TEST_SERVICE_NAME
      systemctl daemon-reload
   fi
```
...with suitable changes in the tests themselves.

Reply:
Hi @edsantiago there is a test case "podman auto-update with multiple services" which request we start several services once inside one test. And I tried to use global variable, but seems bats has it's own way to handle variables. So the original way for transfer variables between functions is not working. We can not change the var inside a function. So I had to store them in a file.